Row index to bucket the ped totals

RyanTG opened the pull request:
I don't expect you to actually merge this. But here's my first take at putting the times into various buckets.

(I had some trouble _really_ concentrating on this, due to the open office environment and coworkers peppering me with questions. But I did my best before the day was through.)

Basically, I just sliced into the array in specific chunks for each time slot (7-10am; 10am-3pm; 3-6pm) and pulled out the totals for each chunk. I did it like this because the "Start Time" column isn't formatted as datetime, so I couldn't split the array by date ranges. And given (/assuming) that the time range is the same for each count, these chunks should be consistent for each count file.

I think this was a fun exercise. And as time permits, and after your feedback, I'll continue digging into it.

cc @calvinbrown32

RyanTG committed:
Row index to bucket the ped totals

aaronfraint commented:
Good work. A few comments though:
1) Don't commit the .ipynb files. I thought this was in the .gitignore. If not, we should add it. Jupytext allows us to treat the .py files like notebooks, and with git we can get a meaningful diff.

2) Although it's likely that all files have the same time buckets, it's dangerous to assume that. As a result I'd suggest converting the timestamp column to a true date time type, and you should be able to run queries against the data frame where you call out actual time values.

3) You now have four "return" lines. Not sure if this works, but convention is to have a single return. If you want to return many things to the user, you'd want to drop them into a list, dictionary, tuple, etc.

RyanTG commented:
Thanks for the feedback! I just added .ipynb to .gitignore (only the checkpoint folder was in there), and removed the file from the project.

I'll address your other points.

I agree that it's dangerous to assume that all files have the same buckets. It took me a little while to explore various methods for bucketing these counts. This first pass was pretty crude, but it helped me learn. I'll work on converting it to date time.

The four returns is something I did at the last second after testing each count individually. So I hadn't gotten a chance to actually test it. Thanks for pointing out that it might not be valid.
